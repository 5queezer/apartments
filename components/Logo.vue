<template>
  <div>
    <div class="PojoniLogo">
      <span class="text-primary">pojoni</span>
      <span class="text-secondary">code challenge</span>
    </div>
  </div>
</template>
<style>
@import url('https://fonts.googleapis.com/css?family=Exo+2:800&display=swap');

.PojoniLogo {
  animation: 1s appear;
  font-family: 'Exo 2', sans-serif;
  font-size: 3em;
}

@keyframes appear {
  0% {
    opacity: 0;
  }
  100% {
    opacity: 1;
  }
}
</style>
