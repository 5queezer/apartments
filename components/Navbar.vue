<template>
  <b-navbar type="light" variant="light" toggleable="lg">
    <b-nav-form>
      <search />
    </b-nav-form>

    <b-navbar-toggle target="nav-collapse">
      <b-nav-text>
        Filter <i class="fa fa-chevron-down" />
      </b-nav-text>
    </b-navbar-toggle>

    <b-collapse id="nav-collapse" is-nav>
      <b-navbar-nav>
        <b-nav-item-dropdown id="dropdown-form-price" text="Price">
          <b-dropdown-form>
            <b-form-row>
              <range
                :values="priceRange"
                append="€"
              />
            </b-form-row>
          </b-dropdown-form>
        </b-nav-item-dropdown>
      </b-navbar-nav>

      <b-navbar-nav>
        <b-nav-item-dropdown text="Area">
          <b-dropdown-form>
            <b-form-row>
              <range
                :values="areaRange"
                append="m²"
              />
            </b-form-row>
          </b-dropdown-form>
        </b-nav-item-dropdown>
      </b-navbar-nav>
      <b-navbar-nav>
        <b-nav-item-dropdown ref="dropdown" text="Bedrooms">
          <b-dropdown-form>
            <b-form-radio-group
              v-model="bedrooms"
              :options="bedroomRange"
              :selected="bedroomRange[0]"
              buttons
            />
          </b-dropdown-form>
        </b-nav-item-dropdown>
      </b-navbar-nav>
      <b-navbar-nav>
        <b-nav-item-dropdown id="dropdown-form4" ref="dropdown" text="Bathroms">
          <b-dropdown-form>
            <b-form-radio-group
              v-model="bathrooms"
              :options="bathroomRange"
              :selected="bathroomRange[0]"
              buttons
            />
          </b-dropdown-form>
        </b-nav-item-dropdown>
      </b-navbar-nav>
    </b-collapse>
  </b-navbar>
</template>

<script>
import _ from 'underscore'
import Search from '~/components/Search.vue'
import Range from '~/components/Range.vue'

export default {
  components: {
    Search,
    Range
  },
  props: {
    priceRange: {
      type: Array,
      default () { return _.range(0, 350e3, 25e3).concat(_.range(350e3, 600e3, 50e3)) }
    },
    areaRange: {
      type: Array,
      default () { return _.range(0, 40, 5).concat(_.range(40, 151, 10)).concat(200, 300) }
    },
    bedroomRange: {
      type: Array,
      default () { return [{ text: 'All', value: undefined }].concat(_.range(1, 6)) }
    },
    bathroomRange: {
      type: Array,
      default () { return [{ text: 'All', value: undefined }].concat(_.range(1, 6)) }
    }
  },
  data () {
    return {
      bedrooms: undefined,
      bathrooms: undefined
    }
  }
}
</script>

<style>
#nav-collapse {
  justify-content: space-around;
}
</style>
