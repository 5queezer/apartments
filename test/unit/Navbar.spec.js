import { createLocalVue, mount, shallowMount } from '@vue/test-utils'
import Bootstrap from 'bootstrap-vue'
import Logo from '@/components/Logo.vue'
import Navbar from '@/components/Navbar.vue'
import Search from '@/components/Search.vue'

const localVue = createLocalVue()
localVue.use(Bootstrap)

describe('Page Elements', () => {
  let navbar
  const options = { localVue }

  test('Navbar', () => {
    navbar = mount(Navbar, options)
    expect(navbar.isVueInstance()).toBeTruthy()
  })

  test('Search field exists', () => {
    const search = navbar.find(Search)
    expect(search.is(Search)).toBe(true)
  })
})
