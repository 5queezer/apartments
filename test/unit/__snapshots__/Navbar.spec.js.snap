// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Page Elements area range matches 1`] = `
Array [
  0,
  5,
  10,
  15,
  20,
  25,
  30,
  35,
  40,
  50,
  60,
  70,
  80,
  90,
  100,
  110,
  120,
  130,
  140,
  150,
  200,
  300,
]
`;

exports[`Page Elements price range matches 1`] = `
Array [
  0,
  25000,
  50000,
  75000,
  100000,
  125000,
  150000,
  175000,
  200000,
  225000,
  250000,
  275000,
  300000,
  325000,
  350000,
  400000,
  450000,
  500000,
  550000,
]
`;
